src/libraries/NormalizedAmount.sol

```diff
@@ -1,58 +1,137 @@
-    return NormalizedAmount.wrap(NormalizedAmount.unwrap(a) + NormalizedAmount.unwrap(b));
-}
+    function getAmount(NormalizedAmount memory a) internal pure returns (uint64) {
+        return a.amount;
+    }
 
-function div(NormalizedAmount a, uint64 b) pure returns (NormalizedAmount) {
-    return NormalizedAmount.wrap(NormalizedAmount.unwrap(a) / b);
-}
+    function getDecimals(NormalizedAmount memory a) internal pure returns (uint8) {
+        return a.decimals;
+    }
 
-function min(NormalizedAmount a, NormalizedAmount b) pure returns (NormalizedAmount) {
-    return a > b ? b : a;
-}
+    function gt(
+        NormalizedAmount memory a,
+        NormalizedAmount memory b
+    ) internal pure returns (bool) {
+        return a.amount > b.amount;
+    }
 
-function unwrap(NormalizedAmount a) pure returns (uint64) {
-    return NormalizedAmount.unwrap(a);
-}
+    function lt(
+        NormalizedAmount memory a,
+        NormalizedAmount memory b
+    ) internal pure returns (bool) {
+        return a.amount < b.amount;
+    }
 
-library NormalizedAmountLib {
-    error AmountTooLarge(uint256 amount);
+    function isZero(NormalizedAmount memory a) internal pure returns (bool) {
+        return (a.amount == 0 && a.decimals == 0);
+    }
 
-    function normalize(uint256 amount, uint8 decimals) internal pure returns (NormalizedAmount) {
-        if (decimals > 8) {
-            amount /= 10 ** (decimals - 8);
+    function sub(
+        NormalizedAmount memory a,
+        NormalizedAmount memory b
+    ) internal pure returns (NormalizedAmount memory) {
+        if (a.amount < b.amount) {
+            revert AmountUnderflows(a.amount, b.amount);
         }
-        // amount after normalization must fit into uint64 (that's the point of
-        // normalization, as Solana only supports uint64 for token amounts)
-        if (amount > type(uint64).max) {
-            revert AmountTooLarge(amount);
+
+        // on initialization
+        if (isZero(b)) {
+            return a;
+        }
+
+        if (a.decimals != b.decimals) {
+            revert NumberOfDecimalsNotEqual(a.decimals, b.decimals);
+        }
+
+        return wrap(a.amount - b.amount, a.decimals);
+    }
+
+    function add(
+        NormalizedAmount memory a,
+        NormalizedAmount memory b
+    ) internal pure returns (NormalizedAmount memory) {
+        // on initialization
+        if (isZero(a)) {
+            return b;
+        }
+
+        if (isZero(b)) {
+            return a;
+        }
+
+        if (a.decimals != b.decimals) {
+            revert NumberOfDecimalsNotEqual(a.decimals, b.decimals);
+        }
+        return wrap(a.amount + b.amount, a.decimals);
+    }
+
+    function min(
+        NormalizedAmount memory a,
+        NormalizedAmount memory b
+    ) public pure returns (NormalizedAmount memory) {
+        return a.amount < b.amount ? a : b;
+    }
+
+    function minDecimals(uint8 toDecimals, uint8 fromDecimals) internal pure returns (uint8) {
```

Comment:
this is just a regular min function on u8s

Reply:
yea afaik there's no `min` function defined on uint8's in the standard library.